cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs

```diff
@@ -667,96 +667,164 @@ fn parse_url(
 mod tests {
     use super::*;
 
-    #[test]
-    fn registry_try_parse_valid() {
-        let tests = vec![
-            (
-                "http://localhost:8080",
-                Registry {
+    struct TestCase {
+        input: String,
+        expected_registry: Registry,
+        expected_registry_host_port_string: String,
+        expected_registry_host_port_namespaced_string: String,
+    }
+
+    fn setup_test_cases() -> Vec<TestCase> {
+        vec![
+            TestCase {
```

Comment:
personally I'm missing some kind of hint about the intent of each testcases - what is usually called "name" in Go tests ;)

I can figure it out if I look at all testcases but if every case had a comment like "hostname with port and path" it'd be easier to understand on a first glance

Reply:
Thanks for the review and the feedback. Let me improve this.